examples/vscode-sandbox/README.md

```diff
@@ -98,19 +98,47 @@ $ kubectl wait --for=condition=Ready sandbox sandbox-example
 $ kubectl get pods -o jsonpath=$'{range .items[*]}{.metadata.name}: {.spec.runtimeClassName}\n{end}'
 ```
 
-## Accesing vscode
+## Accessing VSCode
 
-Port forward the vscode server port.
+To access the VSCode instance securely and reliably, use the Sandbox Router.
 
+### Option 1: Production (via Gateway)
+
+If you have deployed the Gateway **and** Router (see `clients/python/agentic-sandbox-client/sandbox_router`), access VSCode via the external Gateway IP.
+
+1. Get the Gateway IP:
+```bash
+export GATEWAY_IP=$(kubectl get gateway external-http-gateway -n default -o jsonpath='{.status.addresses[0].value}')
+echo "Gateway IP: $GATEWAY_IP"
+```
+
+2. **Connect:** You must inject the routing headers. Use `curl` to test:
+```bash
+# Replace <GATEWAY_IP> with the actual IP
+curl -v -H "X-Sandbox-ID: sandbox-example" \
+        -H "X-Sandbox-Port: 13337" \
+        http://$GATEWAY_IP
 ```
- kubectl port-forward --address 0.0.0.0 pod/sandbox-example 13337
+
+### Option 2: Development (via Router Tunnel)
+
+For local development, port-forward to the **Router Service** (do not port-forward to the pod directly, as it may fail with secure runtimes like gVisor).
```

Comment:
Router is a prerequisite for option 2 as well, right? So the router installation step needs to be called out in option 2 or move up to above both options. 

Also suggest changing the vague "may fail" text here:

```suggestion
For local development, port-forward to the **Router Service** (do not port-forward to the pod directly, as it's not compatible with secure runtimes like gVisor).
```